### PR TITLE
fix pnl for guni

### PR DESCRIPTION
**Pull request by paszkowskiDamian**
# [Title](https://clubhouseLink)

<please insert a clubhouse link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>
- item 1
  
## How to test 🧪
  <Please explain how to test your changes>
- step 1 ...
    
## Definition of done ✔️

- [ ] Acceptance criteria for each issue met
- [ ] Unit tests written where needed and passing
- [ ] End-to-end tests for happy path
- [ ] Documentation updated <When applicable>
- [ ] Project builds without errors
- [ ] Non-functional requirements met
- [ ] Code reviewed and functionality tested by the reviewer (locally, Heroku, etc.)
- [ ] Feature verified and accepted by Product Owner
- [ ] Project deployed to production environment

### CODE REVIEW

File: blockchain/config.ts [hunk 1]
```diff
@@ -234,17 +234,17 @@ const hardhat: NetworkConfig = {
...
+  //   dssMultiplyProxyActions,
+  //   getConfig()?.publicRuntimeConfig?.multiplyProxyActions ||
+  //     '0x2a49eae5cca3f050ebec729cf90cc910fadaf7a2',
+  // ),
+  // // guniProxyActions: contractDesc(guniProxyActions, '0xBEc49fA140aCaA83533fB00A2BB19bDdd0290f25'),
+  // exchange: contractDesc(
+  //   exchange,
+  //   getConfig()?.publicRuntimeConfig?.exchangeAddress ||
+  //     '0x4C4a2f8c81640e47606d3fd77B353E87Ba015584',
+  // ),
```
Comment (piekczyk) [inline blockchain/config.ts:247]: This block of code will no longer be needed or is it a temporary suppression?

**Review by piekczyk — APPROVED** [review]

File: features/openMultiplyVault/variants/guni/open/GuniOpenMultiplyVaultDetails.tsx [hunk 2]
```diff
@@ -98,6 +99,8 @@ export function GuniOpenMultiplyVaultDetails(props: OpenGuniVaultState) {
             showAfterPill,
             currentPnL,
             totalGasSpentUSD,
+            priceInfo,
+            vault: undefined,
```
Comment (piekczyk) [inline features/openMultiplyVault/variants/guni/open/GuniOpenMultiplyVaultDetails.tsx:103]: you could just omit vault here and adjust interface of VaultDetailsCardNetValue:

`vault?: Vault instead` of  `vault: Vault | undefined`

File: features/vaultHistory/VaultHistoryView.tsx [hunk 3]
```diff
@@ -281,18 +282,20 @@ export function VaultHistoryView({ vaultHistory }: { vaultHistory: VaultHistoryE
   const context = useObservable(context$)
   const { t } = useTranslation()
 
+  const spitedEvents = useMemo(() => flatten(vaultHistory.map(splitEvents)), [vaultHistory])
+
```
Comment (piekczyk) [inline features/vaultHistory/VaultHistoryView.tsx:286]: I think that reference of vaultHistory array will be different on each re-render and therefore useMemo hook won't help here. Maybe we could think about using React.memo() with deep equal check. I tried it for this specific scenario and I reduced number of re-renders from 5 to 2 and after that 0 re-renders occurs when I used input field (without memo() this component re-renders on each input change).

File: components/vault/VaultDetails.tsx [hunk 4]
```diff
@@ -685,13 +689,15 @@ export function VaultDetailsCardNetValue({
...
   vault,
+  priceInfo,
 }: {
   netValueUSD: BigNumber
   afterNetValueUSD: BigNumber
   currentPnL: BigNumber
   marketPrice?: BigNumber
   totalGasSpentUSD: BigNumber
-  vault?: Vault
+  vault: Vault | undefined
```
Comment (piekczyk) [inline components/vault/VaultDetails.tsx:699]: vault?: Vault

File: features/openMultiplyVault/variants/guni/open/GuniOpenMultiplyVaultDetails.tsx (hunk 2 above)
Comment (paszkowskiDamian) [reply features/openMultiplyVault/variants/guni/open/GuniOpenMultiplyVaultDetails.tsx:103]: I wanted to be explicit that I don't care about vault in this case. We had bug that in manage we didn't pass vault and Component was fine with it.

**Pull request merged**